### DISCUSSION TITLE
Alignment between geometry column and property columns

### DISCUSSION THREAD

**Discussion by limingchina** [maplibre/maplibre-tile-spec#230] [Q&A]
In the spec, it doesn't mention explicitly about the alignment bewteen the geometry column and property columns. So far, I assume that there is an implicit one-to-one mapping between all the columns. That's also why there is a spec section discussing what would be the best sorting for minimizing the data size. Am I right?

**Comment by nyurik** [comment]
I would let @mactrem to give authoritative answer. I believe this is the case. Moreover, that's why you need "nullable" stream -- a sequence of 0s and 1s to indicate which value is present and which is not -- and the size of that stream also has to match the number of geometries... Although I think one optimization here could be to skip all trailing missing elements - i.e. there could be fewer values in a property column if the last ones are all nulls.

**Comment by mactrem** [comment]
When using feature-scoped properties there is a one-to-one mapping between the id, geometry and property columns for reconstructing a feature.  When stored as vertex-scoped properties then there is a one-to-one mapping of a property to each Vertex in the VertexBuffer of the geometry column, which allows to store so called m-values see
https://github.com/maplibre/maplibre-tile-spec/tree/main/spec#property-columns
